### PR TITLE
TS-4856: Default SSL context fails to load.

### PR DESCRIPTION
**Pull request by shinrich**

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/691/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/795/ for details.

File: iocore/net/SSLUtils.cc [hunk 1]
```diff
@@ -1629,7 +1629,8 @@ SSLInitServerContext(const SSLConfigParams *params, const ssl_user_config *sslMu
     }
 
     if (!sslMultCertSettings->cert) {
-      if (sslMultCertSettings->opt != SSLCertContext::OPT_TUNNEL) {
+      // There must be a certificate specified unless the tunnel action is set, or this is the default case
+      if (sslMultCertSettings->opt != SSLCertContext::OPT_TUNNEL && strcmp(sslMultCertSettings->addr, "*") != 0) {
```
Comment (jpeach) [inline iocore/net/SSLUtils.cc]: In this case could "*" be the default destination configured by the operator? Is that required to have a certificate?

Maybe this validation should be done after parsing a config file entry?

**Comment by shinrich** [comment]
Took @jpeach's suggestion and pulled back a bit.  Moved the missing cert check out of SSLInitServerContext and made the check at the one call site that actually parses the ssl multicert.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/803/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/698/ for details.

**Comment by jpeach** [comment]
👍   This looks pretty reasonable to me.

**Pull request merged**